<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CalculatorDialog</class>
 <widget class="QDialog" name="CalculatorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>548</width>
    <height>517</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Field calculator</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QCheckBox" name="mUpdateExistingFieldCheckBox">
       <property name="text">
        <string>Update existing column</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="mExistingFieldComboBox"/>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="mNewFieldGroupBox">
     <property name="title">
      <string>New Column</string>
     </property>
     <layout class="QGridLayout">
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="mFieldNameLabel">
        <property name="text">
         <string>Name:</string>
        </property>
        <property name="buddy">
         <cstring>mOutputFieldNameLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QLineEdit" name="mOutputFieldNameLineEdit"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QFrame" name="frame">
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="1" rowspan="2" colspan="2">
       <widget class="QFrame" name="mValuesFrame">
        <layout class="QGridLayout">
         <property name="leftMargin">
          <number>9</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>6</number>
         </property>
         <item row="2" column="2">
          <widget class="QPushButton" name="mAllPushButton">
           <property name="text">
            <string>All</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1" rowspan="3">
          <widget class="QListWidget" name="mValueListWidget"/>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QListWidget" name="mFieldsListWidget"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Columns</string>
        </property>
        <property name="textFormat">
         <enum>Qt::AutoText</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Values</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QGroupBox" name="mOperatorsGroupBox">
     <property name="title">
      <string>Operators</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QPushButton" name="mPlusPushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>+</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="mMultiplyPushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>*</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="mSqrtButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>sqrt</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="mEulerButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>exp</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="mMinusPushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="mDividePushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>/</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="mExpButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>^</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="mNaturalLogButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>ln</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="mOpenBracketPushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>(</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="mCloseBracketPushButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>)</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="mFactorialButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>n!</string>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QPushButton" name="mLogButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>log</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QPushButton" name="mNormalButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>rnorm</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="mNumericButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>numeric</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="mRoundButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>round</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="mIfelseButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>ifelse</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="mPoissonButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>rpois</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="mCharacterButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>character</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="mFloorButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>floor</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="mClearButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="mUniformButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>runif</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="mFactorButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>factor</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="mCeilingButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>ceiling</string>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QPushButton" name="mDeleteButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>40</width>
            <height>27</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>80</width>
            <height>27</height>
           </size>
          </property>
          <property name="text">
           <string>Del</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="mFieldCalculatorExpressionLabel">
       <property name="text">
        <string>Expression:</string>
       </property>
       <property name="buddy">
        <cstring>mExpressionTextEdit</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTextEdit" name="mExpressionTextEdit"/>
     </item>
    </layout>
   </item>
   <item row="5" column="0">
    <widget class="QDialogButtonBox" name="mButtonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Help|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>mOutputFieldNameLineEdit</tabstop>
  <tabstop>mValueListWidget</tabstop>
  <tabstop>mAllPushButton</tabstop>
  <tabstop>mPlusPushButton</tabstop>
  <tabstop>mMultiplyPushButton</tabstop>
  <tabstop>mSqrtButton</tabstop>
  <tabstop>mDeleteButton</tabstop>
  <tabstop>mRoundButton</tabstop>
  <tabstop>mCeilingButton</tabstop>
  <tabstop>mOpenBracketPushButton</tabstop>
  <tabstop>mMinusPushButton</tabstop>
  <tabstop>mDividePushButton</tabstop>
  <tabstop>mExpButton</tabstop>
  <tabstop>mNaturalLogButton</tabstop>
  <tabstop>mFloorButton</tabstop>
  <tabstop>mClearButton</tabstop>
  <tabstop>mCloseBracketPushButton</tabstop>
  <tabstop>mNormalButton</tabstop>
  <tabstop>mPoissonButton</tabstop>
  <tabstop>mUniformButton</tabstop>
  <tabstop>mLogButton</tabstop>
  <tabstop>mFactorialButton</tabstop>
  <tabstop>mNumericButton</tabstop>
  <tabstop>mExpressionTextEdit</tabstop>
  <tabstop>mButtonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>mButtonBox</sender>
   <signal>accepted()</signal>
   <receiver>CalculatorDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>679</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mButtonBox</sender>
   <signal>rejected()</signal>
   <receiver>CalculatorDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>679</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
